crates/xilem_web/src/context.rs

```diff
@@ -41,12 +166,50 @@ fn remove_attribute(element: &web_sys::Element, name: &str) {
     }
 }
 
+fn set_class(element: &web_sys::Element, class_name: &str) {
+    debug_assert!(
+        !class_name.is_empty(),
+        "class names cannot be the empty string"
+    );
+    debug_assert!(
+        !class_name.contains(' '),
+        "class names cannot contain the ascii space character"
+    );
+    element.class_list().add_1(class_name).unwrap_throw();
+}
+
+fn remove_class(element: &web_sys::Element, class_name: &str) {
+    debug_assert!(
+        !class_name.is_empty(),
+        "class names cannot be the empty string"
+    );
+    debug_assert!(
+        !class_name.contains(' '),
+        "class names cannot contain the ascii space character"
+    );
+    element.class_list().remove_1(class_name).unwrap_throw();
+}
+
+fn set_style(element: &web_sys::Element, name: &str, value: &str) {
+    // styles will be ignored for non-html elements (e.g. SVG)
+    if let Some(el) = element.dyn_ref::<web_sys::HtmlElement>() {
+        el.style().set_property(name, value).unwrap_throw();
+    }
```

Comment:
```suggestion
    if let Some(el) = element.dyn_ref::<web_sys::HtmlElement>() {
        el.style().set_property(name, value).unwrap_throw();
    } else if let Some(el) = element.dyn_ref::<web_sys::SvgElement>() {
        el.style().set_property(name, value).unwrap_throw();
    }
```

SVGElements do also support style.

Just the base DOM interface (`Element`) doesn't, so to be completely accurate (as e.g. `MathML` AFAIK doesn't support `style`) it would make sense to put `Element::style` in `HtmlElement::style` and `SvgElement::style` (`methods { fn style(..) {..} }` in `dom_interface_macro_and_trait_definitions`)